### DISCUSSION TITLE
get_usage on class based view

### DISCUSSION THREAD

**Discussion by YazidKurdi** [jsocol/django-ratelimit#310] [Q&A]
Hello, I have a class based view that is currently being limited successfully, however I am unable to run get_usage() returning None as the result, is this the correct way of using this method?

```
@method_decorator(ratelimit(key='user_or_ip', rate='2/d',method = 'POST'))
def post(self, request, format=None):
    """
    Generates a recipe based on the provided prompt.
    """

    try:
        print(get_usage(request, group='user_or_ip'))
        # rest of code below
```

Appreciate your response

**Comment by jsocol** [comment]
Hi @YazidKurdi —I believe the issue here is that the `group` keyword argument is not explicitly set on the decorator.

`group` is different from `key`. It is used to distinguish, or group together, different ratelimits across views. For example:

```python
@ratelimit(key='user_or_ip', rate='2/d', group='mygroup')
def view1(request):
    """do stuff here"""

@ratelimit(key='user_or_ip', rate='2/d', group='mygroup')
def view2(request):
    """do other stuff"""
```

When `group` is not explicitly set, `view1` and `view2` will be counted separately, allowing 4 total requests/day between the two of them. Setting both `group` kwargs to the same value means that they share a counter, and 2 requests/day are allowed no matter which view the user is using.

`key` is different, and is used for determining the unique identifiers of the request.

`django-ratelimit` internally creates a unique `group` for each view that doesn't have the kwarg set. In order to be unique, these groups are pretty long and are not a guaranteed/public API (they could change if you changed how you package/deploy your app). For cases like this, setting the `group=` kwarg to the same value on both the `ratelimit()` call and on `get_usage()` should make it possible to get the same count from `get_usage()`. You _may_ need to make sure that the other arguments are also the same, but I'd have to check.

I hope that helps!
